src/kbmod_cmdline/region_searcher.py

```diff
@@ -0,0 +1,402 @@
+"""
+Commandline tool for performing Region Search on an ImageCollection.
+Generates patches, matches them to the ImageCollection, and generates
+ImageCollections for matched patches.
+
+If the output ImageCollections already exist, they will be reused rather than
+regenerated, unless the --overwrite flag is provided.
+
+Also generates analysis tables summarizing the results. Additionally, 
+any errors in processing patches are logged to an errors.csv file in 
+the output directory.
+
+Example Usage
+-------------
+
+The following command takes a base ImageCollection, generates patch grids
+for multiple reflex-correction guess distances and patch sizes, and
+performs Region Search on each configuration. The resulting ImageCollections
+and analysis tables are saved to the specified output directory. 
+
+-------------
+python region_searcher.py \
```

Comment:
I think the convention we have been using for sample command lines is to use ">>>" on the start of the first line (and then indent the following lines) instead of "-------------"